packages/ember-views/lib/components/component.js

```diff
@@ -158,6 +158,22 @@ var Component = View.extend(TargetActionSupport, {
 
       this.layout = this.defaultLayout;
     }
+
+    // If in a tagless component, assert that no event handlers are defined
+    assert(
+      `You can not define a function that handles DOM events in the \`${this}\` tagless component since it doesn't have any DOM element.`,
+      this.tagName !== '' || !(() => {
+        var eventDispatcher = this.container.lookup('event_dispatcher:main');
+
+        for (var key in eventDispatcher._finalEvents) {
```

Comment:
I think we should add a short curcuit for when the `eventDispatcher._finalEvents` is `undefined`.  This might happen when unit testing a component without an event dispatcher setup.